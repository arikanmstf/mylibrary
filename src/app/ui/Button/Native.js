// @flow
import React from 'react';
import { Button as NativeButton, Text } from 'native-base';
import Link from 'ui/Link/Native';

import type { Node } from 'react';
import type { ButtonProps } from './types';

const ButtonLink = (props: ButtonProps): Node => {
  const { to, ...other } = props;

  return (
    <Link
      to={to}
    >
      <Text {...other} />
    </Link>
  );
};

const Button = (props: ButtonProps): Node => {
  const {
    onClick, text, to, ...other
  } = props;

  const foo = () => {};
  const onPress = onClick || foo;
  const Component = to ? ButtonLink : Text;

  return (
    <NativeButton
      full
      onPress={onPress}
      {...other}
    >
      <Component
        to={to}
        {...other}
      >
        {text}
      </Component>
    </NativeButton>
  );
};

export default Button;
