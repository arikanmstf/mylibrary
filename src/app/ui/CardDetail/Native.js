/**
 * Native Component Template By => create-module script
 * @version 1.0.1
 *
 */

// @flow
import React, { Component } from 'react';
import debounce from 'lodash.debounce';
import { withNavigation } from 'react-navigation';
import {
  Share,
  TouchableWithoutFeedback,
  TouchableOpacity,
  ScrollView,
  Linking,
} from 'react-native';
import { Image } from 'ui/native';
import {
  Card,
  CardItem,
  Text,
  Icon,
  Left,
  Body,
  Right,
  ActionSheet,
} from 'native-base';
import { production } from 'constants/routes/createUrl';
import logger from 'helpers/logger';

import { setCardType, defaultProps } from './helpers';
import type { CardDetailProps } from './types';

export class CardDetail extends Component<CardDetailProps> {
  static defaultProps = defaultProps;

  setCardType() {
    const { card, isDetailed } = this.props;
    setCardType.bind(this)(card, isDetailed);
  }

  goToDetail = () => {
    const { card, navigation } = this.props;

    if (card && this.getDetailUrl) {
      logger.log('goToDetail');
      const url = this.getDetailUrl();
      navigation.navigate(url, { id: card.id });
    }
  };

  shareCard = () => {
    const { card } = this.props;
    Share.share({
      url: production()(this.getDetailUrl(card.id)),
    });
  };

  goTo = (option) => {
    const { navigation } = this.props;
    if (option.to) {
      navigation.navigate(option.to, { id: option.toId });
    }
  };

  goToAddToList = () => {
    const { card, navigation } = this.props;

    logger.log('goToAddToList');
    const url = this.addToListUrl();
    navigation.navigate(url, { id: card.id });
  };

  goToDownload = async () => {
    const { card } = this.props;
    if (card && card.downloadUrl) {
      const supported = await Linking.canOpenURL(card.downloadUrl);

      if (supported) {
        Linking.openURL(card.downloadUrl);
      } else {
        logger.log(`Can't open the download url: ${card.downloadUrl}`);
      }
    }
  };

  handleRenderMoreClick = () => {
    const options = this.moreOptions.map((option) => (option.label));
    ActionSheet.show(
      {
        options,
        cancelButtonIndex: options.length - 1,
        destructiveButtonIndex: options.length - 1,
      },
      (buttonIndex) => {
        const option = this.moreOptions[buttonIndex];
        this.goTo(option);
      }
    );
  };

  toggleFavorite(id: number) {
    const { toggleFavorite } = this.props;
    if (toggleFavorite) {
      toggleFavorite(id);
    }
  }

  toggleRead(id: number) {
    const { toggleRead } = this.props;
    if (toggleRead) {
      toggleRead(id);
    }
  }

  render() {
    const { card, isDetailed } = this.props;

    if (!card) {
      return null;
    }
    this.setCardType();
    const goToDetailDebounce = debounce(this.goToDetail, 800, { leading: true, trailing: false });
    const goToAddToListDebounce = debounce(this.goToAddToList, 800, { leading: true, trailing: false });
    const goToDownloadDebounce = debounce(this.goToDownload, 800, { leading: true, trailing: false });

    logger.log('render: CardDetail');

    return (
      <ScrollView>
        <Card>
          <CardItem>
            <Body>
              <Text>{card.title}</Text>
              <Text note>{card.subHeader}</Text>
            </Body>
            <Right
              style={{
                flex: 1,
                alignItems: 'center',
                flexDirection: 'row',
                justifyContent: 'flex-end',
              }}
            >
              { card && this.moreOptions.length > 1 ? (
                <TouchableOpacity
                  onPress={this.handleRenderMoreClick}
                  style={{ width: 40, flexShrink: 1, alignItems: 'flex-end' }}
                >
                  <Icon name="more" active />
                </TouchableOpacity>
              ) : null }
            </Right>
          </CardItem>
          { this.imageUri ? (
            <TouchableWithoutFeedback
              onPress={goToDetailDebounce}
            >
              <CardItem
                cardBody
              >
                <Image
                  source={{ uri: this.imageUri }}
                  style={{ height: 200, flex: 1 }}
                />
              </CardItem>
            </TouchableWithoutFeedback>
          ) : null }
          { isDetailed ? (
            <CardItem>
              <Body>
                <Text>{card.description}</Text>
              </Body>
            </CardItem>
          ) : null }
          <CardItem>
            <Left>
              { card.isFavorite !== undefined ? (
                <TouchableOpacity
                  style={{ width: 40 }}
                  onPress={() => { this.toggleFavorite(card.id); }}
                >
                  <Icon name="star" active={card.isFavorite} style={{ fontSize: 30 }} />
                </TouchableOpacity>
              ) : null }
              { card.isRead !== undefined ? (
                <TouchableOpacity
                  style={{ width: 40, marginTop: 3 }}
                  onPress={() => { this.toggleRead(card.id); }}
                >
                  <Icon name="book" active={card.isRead} style={{ fontSize: 30 }} />
                </TouchableOpacity>
              ) : null }
              { this.addToListUrl ? (
                <TouchableOpacity
                  style={{ width: 40, marginTop: 3 }}
                  onPress={goToAddToListDebounce}
                >
                  <Icon name="add" style={{ fontSize: 30 }} />
                </TouchableOpacity>
              ) : null }
            </Left>
            <Right
              style={{
                flex: 1,
                alignItems: 'center',
                flexDirection: 'row',
                justifyContent: 'flex-end',
              }}
            >
              { card.downloadUrl ? (
                <TouchableOpacity
                  style={{ width: 40 }}
                  onPress={goToDownloadDebounce}
                >
                  <Icon name="download" color="#000" style={{ fontSize: 30 }} />
                </TouchableOpacity>
              ) : null }
              { this.getDetailUrl ? (
                <TouchableOpacity
                  style={{ width: 40, flexShrink: 1, alignItems: 'flex-end' }}
                  onPress={this.shareCard}
                >
                  <Icon name="share" color="#000" style={{ fontSize: 30 }} />
                </TouchableOpacity>
              ) : null }
            </Right>
          </CardItem>
        </Card>
      </ScrollView>
    );
  }
}

export default withNavigation(CardDetail);
