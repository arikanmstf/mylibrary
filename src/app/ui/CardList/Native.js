/**
 * Native Component Template By => create-module script
 * @version 1.0.0
 *
 */

// @flow
import * as React from 'react';
import debounce from 'lodash.debounce';
import { withRouter } from 'react-router-native';
import { connect } from 'react-redux';
import {
  FlatList,
  RefreshControl,
} from 'react-native';
import logger from 'helpers/logger';
import { CardDetail } from 'ui/native';

import { mapStateToProps, mapDispatchToProps } from './actions';
import type { CardListProps, RenderCardListItem } from './types';

const isCloseToBottom = ({ layoutMeasurement, contentOffset, contentSize }) => {
  const paddingToBottom = 600;
  return layoutMeasurement.height + contentOffset.y >= contentSize.height - paddingToBottom;
};

export class CardList extends React.Component<CardListProps> {
  handleScroll = ({ nativeEvent }) => {
    const {
      addCards,
    } = this.props;

    if (isCloseToBottom(nativeEvent)) {
      if (addCards) {
        addCards();
        logger.log('addCards');
      }
    }
  };

  handleRefresh = () => {
    const {
      fetchCards,
      isLoaderVisible,
      search,
    } = this.props;

    if (fetchCards && !isLoaderVisible) {
      fetchCards({ search });
      logger.log('fetchCards');
    }
  };

  renderCardList = (item: RenderCardListItem) => {
    const card = item ? item.item : null;
    return card && (
      <CardDetail
        card={card}
        index={item.index}
        key={card.id}
      />
    );
  };

  render() {
    const { cards } = this.props;
    const handleScrollDebounce = ({ nativeEvent }) => {
      debounce(this.handleScroll, 800, {
        leading: true,
      })({ nativeEvent });
    };

    logger.log('render: CardList');

    return (
      <FlatList
        data={cards}
        renderItem={this.renderCardList}
        onScrollEndDrag={handleScrollDebounce}
        refreshControl={<RefreshControl refreshing={false} onRefresh={this.handleRefresh} />}
      />
    );
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(withRouter(CardList));
