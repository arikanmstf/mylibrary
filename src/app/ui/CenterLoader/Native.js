/**
 * Native Component Template By => create-module script
 * @version 1.0.1
 *
 */

// @flow
import React from 'react';
import type { Node } from 'react';
import { connect } from 'react-redux';
import { Body } from 'native-base';
import { Image } from 'ui/native';
import Loading from 'assets/images/loading.gif';
import { mapStateToProps } from './actions';

import type { CenterLoaderProps } from './types';

const CenterLoader = ({ isVisible }: CenterLoaderProps): Node => {
  return isVisible ? (
    <Body style={{ textAlign: 'center' }}>
      <Image source={Loading} style={{ outline: 'none', width: 25, height: 25 }} />
    </Body>
  ) : null;
};

export default connect(mapStateToProps)(CenterLoader);
