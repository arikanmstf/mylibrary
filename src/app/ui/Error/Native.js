// @flow
import React, { Component, ErrorInfo } from 'react';
import logger from 'helpers/logger';
import { Screen } from 'ui/native';

import type { ErrorProps, ErrorState } from './types';

class ErrorBoundary extends Component<ErrorProps, ErrorState> {
  constructor(props) {
    super(props);
    this.state = { error: null };
  }

  componentDidCatch(error: Error, info: ErrorInfo): void {
    logger.log(error, info);
    this.setState({ error });
  }

  render() {
    const { error } = this.state;
    const { children } = this.props;

    return error && error.stack ? (
      <Screen>
        {error.stack}
      </Screen>
    ) : children;
  }
}

export default ErrorBoundary;
