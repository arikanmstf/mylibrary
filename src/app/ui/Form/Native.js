/**
 * Native Component Template By => create-module script
 * @version 1.0.0
 *
 */

// @flow
import React from 'react';
import { Row } from 'ui/native';

import type { Node } from 'react';
import type { FormProps } from './types';

const Form = (props: FormProps): ?Node => {
  return (<Row {...props} />);
};

export default Form;
