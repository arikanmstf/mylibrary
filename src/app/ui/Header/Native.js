/**
 * Native Component Template By => create-module script
 * @version 1.0.0
 *
 */

// @flow
import React, { PureComponent } from 'react';
import { connect } from 'react-redux';
import { TouchableOpacity } from 'react-native';
import { reduxForm } from 'redux-form/immutable';
import { withNavigation } from 'react-navigation';
import debounce from 'lodash.debounce';
import {
  Header as HeaderNative,
  Item,
  Body,
  Left,
  Right,
  Icon,
  Content,
  List,
  ListItem,
  Text,
  Title,
} from 'native-base';
import { TextField, Link } from 'ui/native';
import t from 'helpers/i18n/Translate';
import logger from 'helpers/logger';
import fields, { SEARCH_FORM_KEY } from 'constants/forms/search';
import { white } from 'constants/theme/color';

import { mapDispatchToProps, submitSearchForm } from './actions';
import routes from './sideNavigationItems';
import { SEARCH_SUBMIT_TIMEOUT } from './types';

import type { HeaderProps, HeaderState, SideNavigationItem } from './types';

export const renderList = (route: SideNavigationItem) => {
  return (
    <ListItem
      button
    >
      <Icon name={route.icon} style={{ marginRight: 20 }} />
      <Link to={route.to}>
        <Text>{t.get(route.label)}</Text>
      </Link>
    </ListItem>
  );
};

export const SideBar = () => (
  <Content style={{ backgroundColor: white, paddingTop: 20 }}>
    <List
      dataArray={routes}
      renderRow={renderList}
    />
  </Content>
);

export class Header extends PureComponent<HeaderProps, HeaderState> {
  state = {
    isSearchFocus: false,
  };

  handleMenuButtonPress = () => {
    const { showDrawer } = this.props;
    if (showDrawer) {
      logger.log('showDrawer');
      showDrawer();
    }
  };

  handleBackButtonPress = () => {
    const { navigation } = this.props;
    logger.log('handleBackButtonPress');

    if (navigation) {
      navigation.goBack();
    }
  };

  handleChange = () => {
    const {
      handleSubmit,
    } = this.props;
    console.log(this.props);

    handleSubmit(submitSearchForm)();
  };

  handleFocus = () => {
    this.setState({
      isSearchFocus: true,
    });
  };

  handleBlur = () => {
    this.setState({
      isSearchFocus: false,
    });
  };

  renderMenu() {
    return (
      <Right>
        <TouchableOpacity
          onPress={this.handleMenuButtonPress}
          style={{ width: 70, flex: 1, alignItems: 'flex-end' }}
        >
          <Icon name="menu" />
        </TouchableOpacity>
      </Right>
    );
  }

  renderBackButton() {
    const { isSearchFocus } = this.state;
    return !isSearchFocus ? (
      <Left>
        <TouchableOpacity
          onPress={this.handleBackButtonPress}
          style={{ width: 70 }}
        >
          <Icon name="arrow-back" />
        </TouchableOpacity>
      </Left>
    ) : null;
  }

  renderCenter() {
    const { title } = this.props;

    return title ? (<Body><Title>{title}</Title></Body>) : (
      <Item>
        <Icon name="ios-search" />
        <TextField
          name={fields.SEARCH}
          placeholder={t.get('HEADER_SEARCH')}
          type="search"
          onChange={debounce(this.handleChange, SEARCH_SUBMIT_TIMEOUT)}
          onFocus={this.handleFocus}
          onBlur={this.handleBlur}
        />
      </Item>
    );
  }

  render() {
    return (
      <HeaderNative searchBar rounded>
        {this.renderBackButton()}
        {this.renderCenter()}
        {this.renderMenu()}
      </HeaderNative>
    );
  }
}

export default reduxForm({
  form: SEARCH_FORM_KEY,
  onSubmit: submitSearchForm,
})(
  connect(null, mapDispatchToProps)(withNavigation(Header))
);
