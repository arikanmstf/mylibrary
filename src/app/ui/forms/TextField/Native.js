/**
 * Native Component Template By => create-module script
 * @version 1.0.0
 *
 */

// @flow
import React from 'react';
import { StyleSheet, TextInput } from 'react-native';
import { Field } from 'redux-form/immutable';
import type { Node } from 'react';

import defaultStyle from './style.json';

import type { TextFieldProps, ReduxFieldProps } from './types';

const renderTextField = ({ input, ...other }: ReduxFieldProps) => (
  <TextInput
    onChangeText={input.onChange}
    {...input}
    {...other}
  />
);

const TextField = (props: TextFieldProps): ?Node => {
  const {
    label,
    style,
    type,
    ...other
  } = props;

  const mergedStyles = StyleSheet.create({
    container: {
      ...defaultStyle,
      ...style,
    },
  });
  const isPassword = type === 'password';

  return (
    <Field
      component={renderTextField}
      placeholder={label}
      style={mergedStyles.container}
      {...other}
      secureTextEntry={isPassword}
    />
  );
};

export default TextField;
