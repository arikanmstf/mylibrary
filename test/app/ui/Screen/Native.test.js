/**
 * Web Component Test Template By => create-module script
 * @version 1.0.0
 *
 */

import React from 'react';
import { shallow } from 'enzyme';
import { Screen } from 'ui/Screen/Native';

describe('test/app/ui/Screen/Native.test.js', () => {
  describe('Render', () => {
    it('props: {}', () => {
      const props = {};
      const wrapper = shallow(<Screen {...props} />);
      expect(wrapper).toMatchSnapshot();
    });

    it('props: { center: true }', () => {
      const props = { center: true };
      const wrapper = shallow(<Screen {...props} />);
      expect(wrapper).toMatchSnapshot();
    });
  });

  it('hideDrawer', async () => {
    const hideDrawer = jest.fn();
    const props = {
      hideDrawer,
    };

    const wrapper = shallow(<Screen {...props} />);
    await wrapper.instance().hideDrawer();
    expect(hideDrawer).toBeCalled();
  });
});
