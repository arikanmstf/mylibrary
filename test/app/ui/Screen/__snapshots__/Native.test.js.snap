// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test/app/ui/Screen/Native.test.js Render props: { center: true } 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Screen
    center={true}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <ScrollView
        contentContainerStyle={
          Object {
            "alignItems": "center",
            "backgroundColor": "#fffffc",
            "flex": 1,
            "justifyContent": "center",
          }
        }
        keyboardShouldPersistTaps="handled"
        scrollEnabled={false}
      />,
      "closedDrawerOffset": 0,
      "content": <SideBar />,
      "onClose": [Function],
      "open": undefined,
      "openDrawerOffset": 0.2,
      "panCloseMask": 0.2,
      "side": "right",
      "styles": Object {
        "drawer": Object {
          "elevation": 5,
          "shadowColor": "#000000",
          "shadowOpacity": 0,
          "shadowRadius": 0,
        },
        "mainOverlay": Object {
          "backgroundColor": "rgba(0, 0, 0, 0.8)",
          "elevation": 8,
          "opacity": 0,
        },
      },
      "tapToClose": true,
      "tweenHandler": [Function],
      "type": "overlay",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "contentContainerStyle": Object {
          "alignItems": "center",
          "backgroundColor": "#fffffc",
          "flex": 1,
          "justifyContent": "center",
        },
        "keyboardShouldPersistTaps": "handled",
        "scrollEnabled": false,
      },
      "ref": null,
      "rendered": null,
      "type": [MockFunction],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <ScrollView
          contentContainerStyle={
            Object {
              "alignItems": "center",
              "backgroundColor": "#fffffc",
              "flex": 1,
              "justifyContent": "center",
            }
          }
          keyboardShouldPersistTaps="handled"
          scrollEnabled={false}
        />,
        "closedDrawerOffset": 0,
        "content": <SideBar />,
        "onClose": [Function],
        "open": undefined,
        "openDrawerOffset": 0.2,
        "panCloseMask": 0.2,
        "side": "right",
        "styles": Object {
          "drawer": Object {
            "elevation": 5,
            "shadowColor": "#000000",
            "shadowOpacity": 0,
            "shadowRadius": 0,
          },
          "mainOverlay": Object {
            "backgroundColor": "rgba(0, 0, 0, 0.8)",
            "elevation": 8,
            "opacity": 0,
          },
        },
        "tapToClose": true,
        "tweenHandler": [Function],
        "type": "overlay",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "contentContainerStyle": Object {
            "alignItems": "center",
            "backgroundColor": "#fffffc",
            "flex": 1,
            "justifyContent": "center",
          },
          "keyboardShouldPersistTaps": "handled",
          "scrollEnabled": false,
        },
        "ref": null,
        "rendered": null,
        "type": [MockFunction],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`test/app/ui/Screen/Native.test.js Render props: {} 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Screen />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <ScrollView
        contentContainerStyle={
          Object {
            "backgroundColor": "#fffffc",
            "flex": 1,
          }
        }
        keyboardShouldPersistTaps="handled"
        scrollEnabled={false}
      />,
      "closedDrawerOffset": 0,
      "content": <SideBar />,
      "onClose": [Function],
      "open": undefined,
      "openDrawerOffset": 0.2,
      "panCloseMask": 0.2,
      "side": "right",
      "styles": Object {
        "drawer": Object {
          "elevation": 5,
          "shadowColor": "#000000",
          "shadowOpacity": 0,
          "shadowRadius": 0,
        },
        "mainOverlay": Object {
          "backgroundColor": "rgba(0, 0, 0, 0.8)",
          "elevation": 8,
          "opacity": 0,
        },
      },
      "tapToClose": true,
      "tweenHandler": [Function],
      "type": "overlay",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "contentContainerStyle": Object {
          "backgroundColor": "#fffffc",
          "flex": 1,
        },
        "keyboardShouldPersistTaps": "handled",
        "scrollEnabled": false,
      },
      "ref": null,
      "rendered": null,
      "type": [MockFunction],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <ScrollView
          contentContainerStyle={
            Object {
              "backgroundColor": "#fffffc",
              "flex": 1,
            }
          }
          keyboardShouldPersistTaps="handled"
          scrollEnabled={false}
        />,
        "closedDrawerOffset": 0,
        "content": <SideBar />,
        "onClose": [Function],
        "open": undefined,
        "openDrawerOffset": 0.2,
        "panCloseMask": 0.2,
        "side": "right",
        "styles": Object {
          "drawer": Object {
            "elevation": 5,
            "shadowColor": "#000000",
            "shadowOpacity": 0,
            "shadowRadius": 0,
          },
          "mainOverlay": Object {
            "backgroundColor": "rgba(0, 0, 0, 0.8)",
            "elevation": 8,
            "opacity": 0,
          },
        },
        "tapToClose": true,
        "tweenHandler": [Function],
        "type": "overlay",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "contentContainerStyle": Object {
            "backgroundColor": "#fffffc",
            "flex": 1,
          },
          "keyboardShouldPersistTaps": "handled",
          "scrollEnabled": false,
        },
        "ref": null,
        "rendered": null,
        "type": [MockFunction],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
